### PR TITLE
healthcheck: Cache regions in tablet_stats_cache

### PR DESCRIPTION
**Pull request by leoxlin**
This adds caches getRegionByCell call to improve initial vtgate healthcheck performance.

We've seen some significant initial healthcheck slow down in our recent vitess upgrade that resulted in the gateway timing out before finishing waiting for the `-tablet_types_to_wait`

We were able to diagnose this via a lock contention profile that showed `StatsUpdate` to be the pain point. From there @sougou was able to identify the uncached cell region call.

Signed-off-by: Leo Xuzhang Lin <llin@hubspot.com>

### CODE REVIEW

**Comment by leoxlin** [comment]
Context: https://vitess.slack.com/archives/C0PQY0PTK/p1548797459719600

File: go/vt/discovery/tablet_stats_cache.go [hunk 1]
```diff
@@ -193,13 +196,36 @@ func (tc *TabletStatsCache) getOrCreateEntry(target *querypb.Target) *tabletStat
...
+	defer tc.mu.RUnlock()
+	if region, ok := tc.cellRegions[cell]; ok {
+		return region, true
+	}
+	return "", false
+}
+
+func (tc *TabletStatsCache) getOrCreateRegionByCell(cell string) string {
+	// Fast path
+	if region, ok := tc.getRegionByCell(cell); ok {
```
Comment (sougou) [inline go/vt/discovery/tablet_stats_cache.go]: No need for a separate fast path. You can just check of tc.cellRegions inside the 'slow' path. It's essentially the same thing. Which means you don't need a separate `getRegionByCell` function. Which means you could just rename this function to `getRegionByCell`.

**Comment by sougou** [comment]
Also, there's a new test failure.

**Comment by leoxlin** [comment]
I've addressed all comments and fixed the test

**Pull request merged**